EIPS/eip-4362.md

```diff
@@ -0,0 +1,270 @@
+- SIWE Client (SC) - The entity initiating the SIWE authentication and CapGrok flow.
+- Relying Party (RP) - same as SC in the context of authentication.
+
+### Overview
+
+This specification defines the following:
+- CapGrok SIWE Extension
+- CapGrok Capability
+   - CapGrok URI Scheme
+   - CapGrok Details Object Schema
+- CapGrok Translation Algorithm
+- CapGrok Verification
+
+### CapGrok SIWE Extension
+
+A CapGrok is a EIP-4361 message following a specific format that allows an Ethereum account to delegate a set of CapGrok Capabilities to a delegee through informed consent. Each CapGrok Capability MUST be represented by an entry in the `Resources` array of the SIWE message that MUST deterministically translate the CapGrok Capability in human-readable form to the `statement` field in the SIWE message using the CapGrok Translation Algorithm.
+
+The following SIWE message fields are used to further define (or limit) the scope of all CapGrok Capabilities:
+- The `URI` field MUST specify the delegee as a URI, e.g., `https://example.com`, `did:key:z6MkhaXgBZDvotDkL5257faiztiGiC2QtKLGpbnnEGta2doK`. It is expected that the RS authenticates the delegee before invoking an action for the CapGrok Capability.
+- The `Issued At` field MUST be used to specify the issuance date of the CapGrok Capabilities.
+- If present, the `Expiration Time` field MUST be used as the expiration time of the CapGrok Capabilities where the RS no longer accepts invoking the associated actions.
+- If present, the `Not Before` field MUST be used as the time that has to expire before the RS starts accepting invoking of the associated actions of the CapGrok Capabilities.
+
+The following is a non-normative example of a SIWE message with the SIWE CapGrok Extension:
+```text
+example.com wants you to sign in with your Ethereum account:
+0x0000000000000000000000000000000000000000
+
+I further authorize https://example.com to perform the following actions on my behalf: (1) example: read for any. (2) example: append, delete for my.resource.1. (3) example: append for my.resource.2, my.resource.3.
+
+URI: https://example.com
+Version: 1
+Chain ID: 1
+Nonce: n-0S6_WzA2Mj
+Issued At: 2022-06-21T12:00:00.000Z
+Resources:
+- urn:capability:example:eyJkZWYiOlsicmVhZCJdLCJ0YXIiOnsibXkucmVzb3VyY2UuMSI6WyJhcHBlbmQiLCJkZWxldGUiXSwibXkucmVzb3VyY2UuMiI6WyJhcHBlbmQiXSwibXkucmVzb3VyY2UuMyI6WyJhcHBlbmQiXX19
+```
+
+#### CapGrok Capability
+
+A CapGrok Capability is identified by their CapGrok URI that resolves to a CapGrok Details Object which defines the associated actions and optional target resources. The scope of each CapGrok Capability is attenuated by common fields in the SWIE message as described in the previous chapter, e.g., `URI`, `Issued At`, `Expiration Time`, `Not Before`.
+
+##### CapGrok URI Scheme
+
+A CapGrok URI starts with `urn:capability:` followed by the namespace discriminator, followed by `:` and the base64url-encoded payload of the CapGrok Details Object.
```

Comment:
What is `base64url`? You _cannot_ link to its definition directly, but it might be worthwhile either including a copy of the spec (if it has a compatible license) in the `/assets/eip-5573` folder.

Reply:
@SamWilsn It is defined in RFC4648 --> chapter 5 defines base64url. I updated the PR to mention where base64url is defined.

Reply:
@SamWilsn pls let me know if this works for you.